src/mono/wasm/runtime/library_mono.js

```diff
@@ -536,109 +535,371 @@ var MonoSupportLib = {
-							} catch (ex) {
-								print ("MONO_WASM: load_runtime () failed: " + ex);
-								var err = new Error();
-								print ("MONO_WASM: Stacktrace: \n");
-								print (err.stack);
-
-								var wasm_exit = Module.cwrap ('mono_wasm_exit', null, ['number']);
-								wasm_exit (1);
+				};
+
+				attemptNextSource = function () {
+					if (sourceIndex >= sourcesList.length) {
+						var msg = "MONO_WASM: Failed to load " + asset.name;
+						try {
+							var isOk = asset.is_optional ||
+								(asset.name.match (/\.pdb$/) && MONO.mono_wasm_ignore_pdb_load_errors);
+
+							if (isOk)
+								console.log (msg);
+							else {
+								console.error (msg);
+								throw new Error (msg);
 							}
+						} finally {
+							onPendingRequestComplete ();
+						}
+					}
+
+					var sourcePrefix = sourcesList[sourceIndex];
+					sourceIndex++;
+
+					// HACK: Special-case because MSBuild doesn't allow "" as an attribute
+					if (sourcePrefix === "./")
+						sourcePrefix = "";
+
+					var attemptUrl;
+					if (sourcePrefix.trim() === "") {
+						if (asset.behavior === "assembly")
+							attemptUrl = locateFile (args.assembly_root + "/" + asset.name);
+						else
+							attemptUrl = asset.name;
+					} else {
+						attemptUrl = sourcePrefix + asset.name;
+					}
+
+					try {
+						if (asset.name === attemptUrl) {
+							if (ctx.tracing)
+								console.log ("Attempting to fetch '" + attemptUrl + "'");
 						} else {
-							load_runtime (vfs_prefix, enable_debugging);
+							if (ctx.tracing)
+								console.log ("Attempting to fetch '" + attemptUrl + "' for", asset.name);
 						}
-						MONO.mono_wasm_runtime_ready ();
-						loaded_cb ();
+						var fetch_promise = fetch_file_cb (attemptUrl);
+						fetch_promise.then (handleFetchResponse);
+					} catch (exc) {
+						console.error ("MONO_WASM: Error fetching " + attemptUrl, exc);
+						attemptNextSource ();
 					}
-				});
+				};
+
+				attemptNextSource ();
 			});
 		},
 
+		_globalization_init: function (args, ctx) {
+			var invariantMode = false;
+
+			if (args.globalization_mode === "invariant")
+				invariantMode = true;
+
+			if (!invariantMode) {
+				if (ctx.num_icu_assets_loaded_successfully > 0) {
+					console.log ("MONO_WASM: ICU data archive(s) loaded, disabling invariant mode");
+				} else if (args.globalization_mode !== "icu") {
+					console.log ("MONO_WASM: ICU data archive(s) not loaded, using invariant globalization mode");
+					invariantMode = true;
+				} else {
+					var msg = "invariant globalization mode is inactive and no ICU data archives were loaded";
+					console.error ("MONO_WASM: ERROR: " + msg);
+					throw new Error (msg);
+				}
+			}
+
+			if (invariantMode)
+				this.mono_wasm_setenv ("DOTNET_SYSTEM_GLOBALIZATION_INVARIANT", "1");
+		},
+
 		mono_wasm_get_loaded_files: function() {
-			console.log(">>>mono_wasm_get_loaded_files");
-			return this.loaded_files;
+			return Object.keys(MONO.loaded_assets);
```

Comment:
Actually this won't work, this list should consist of only the assemblies and pdb and this method needs to actually return MONO.loaded_files for now

Reply:
MONO.loaded_files is gone, we'll need to build a list of assemblies only then

Reply:
https://github.com/dotnet/aspnetcore/blob/6cb575216f00a33801711ab7e91259f20554f848/src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts#L383

Reply:
you can populate MONO.loaded_files when you add the assemblies to the runtime just like that.  We can discuss a better fix later but I'd like to avoid breaking things right now.